src/module.js

```diff
@@ -34,74 +37,121 @@ const execute = (command, log) => {
 
         return;
     }
 
     this
         .on('kill', (done) => {
             if (kill) {
-                return execute(`VBoxManage controlvm {${ uuid }} acpipowerbutton`, log)
-                    .then(() => done())
-                    .catch(() => {
-                        log.error('Failed to turn of the vitual machine.');
-
-                        done();
+                uuidFromVMName(args.vmName, args.uuid, log)
+                    .then((uuid) => {
+                        return execute(`VBoxManage controlvm {${ uuid }} acpipowerbutton`, log)
+                            .then(() => done())
+                            .catch(() => {
+                                log.error('Failed to turn of the vitual machine.');
+
+                                done();
+                            });
+                    })
+                    .catch((err) => {
+                        log.error(err);
                     });
             }
 
             done();
         })
         .on('start', (url) => {
-            execute('VBoxManage list runningvms', log)
-                .then((result) => {
-                    if (!result.includes(`{${ uuid }}`)) {
-                        let queue;
-
-                        if (snapshot === undefined) {
-                            queue = Promise.resolve();
-                        } else {
-                            queue = execute(`VBoxManage snapshot {${ uuid }} restore "${ snapshot }"`, log);
-                        }
-
-                        return queue
-                            .then(() => execute(`VBoxManage startvm {${ uuid }}`, log))
-                            // Wait for the LoggedInUsers to become 0.
-                            .then(() => execute(`VBoxManage guestproperty wait {${ uuid }} /VirtualBox/GuestInfo/OS/LoggedInUsers --timeout 1800000`, log))
-                            // Wait for the LoggedInUsers to become 1.
-                            .then(() => execute(`VBoxManage guestproperty wait {${ uuid }} /VirtualBox/GuestInfo/OS/LoggedInUsers --timeout 1800000`, log))
-                            .then(() => new Promise((resolve) => {
-                                // Wait one more minute to be sure that Windows is up and running.
-                                setTimeout(resolve, 60e3);
-                            }));
-                    }
-
-                    log.info('The virtual machine is already running.');
-                })
-                .then(() => {
-                    return execute(`VBoxManage guestcontrol {${ uuid }} --password Passw0rd! --username IEUser run --exe "C:\\Program Files\\Internet Explorer\\iexplore.exe" --wait-stderr --wait-stdout -- -extoff -private ${ url.replace(/localhost:/, '10.0.2.2:') }`, log);
+            uuidFromVMName(args.vmName, args.uuid, log)
+                .then((uuid) => {
+                    return execute('VBoxManage list runningvms', log)
```

Comment:
I left the original implementation of .on('start') untouched, except for wrapping it inside the call to `uuidFromVMName`. The only other differences here should be just whitespace.